big-o-notation/big-o.js

```diff
@@ -0,0 +1,47 @@
+/* exported uniqueQuadratic, uniqueLinear */
+
+function uniqueLinear(words) {
+  const seen = {}; // 1 * 1= O(1)
+  const unique = []; // 1 * 1= O(1)
+  for (
+    let i = 0; // 1 * 1 = O(1)
+    i < words.length; // 2 * n= O(n)
+    i++ // 1 * n = O(1n)
+  ) {
+    const word = words[i]; // 2 * n = O(n)
+    if (!seen[word]) {
+      // n  * 2 = O(n)
+      seen[word] = true; // n * 1 = O(n)
+      unique[unique.length] = word; // n * 1 = O(n)
+    }
+  }
+  return unique; // 1 * 1= O(1)
+} // Big O Notation for uniqueLinear: O(n)
+
+function uniqueQuadratic(words) {
+  const unique = []; // 1 * 1 = O(1)
+  for (
+    let i = 0; // 1 * 1= O(1)
+    i < words.length; // 2 * n = O(n)
+    i++ // 1 * n = O(?)
+  ) {
+    const word = words[i]; // 2 * n = O(n)
+    let isUnique = true; // 1 * 1 = O(1)
+    for (
+      let c = 0; // 1 * 1 = O(1)
+      c < i; // 1 * 1 = O(1)
```

Comment:
These lines here all execute once each time through this inner loop. The inner loop executes once each time through the outer loop. So they execute n*n times, which gives O(n^2).